Rebus/CloudEventify.Rebus.Tests/SerializerSourceAddressTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using AutoFixture;
+using CloudNative.CloudEvents;
+using FluentAssertions;
+using Moq;
+using Rebus.Config;
+using Rebus.Messages;
+using Rebus.Serialization;
+using Rebus.Serialization.Custom;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace CloudEventify.Rebus.Tests
+{
+    public class SerializerSourceAddressTests
+    {
+        private CloudEventFormatter _formatter = Formatter.New(new JsonSerializerOptions());
+        private Mock<IMessageTypeNameConvention> _messageTypeNameConventionMock;
+        private Message _simpleBaseMessage;
+
+        public SerializerSourceAddressTests()
+        {
+            _messageTypeNameConventionMock = new Mock<IMessageTypeNameConvention>();
+            _messageTypeNameConventionMock.Setup(m => m.GetTypeName(It.IsAny<Type>())).Returns("thing");
+            _simpleBaseMessage = new Fixture().Build<Message>().Create();
+            _simpleBaseMessage.Headers.Clear();
+            _simpleBaseMessage.Headers.Add(Headers.MessageId, Guid.NewGuid().ToString());
+            _simpleBaseMessage.Headers.Add(Headers.SentTime, DateTimeOffset.UtcNow.ToString("O"));
+            _simpleBaseMessage.Headers.Add(Headers.Type, "thing");
+            _simpleBaseMessage.Headers.Add(Headers.SenderAddress, "things");
```

Comment:
I would like you to put this into a factory method instead of the constructor initializing a field

Reply:
That is, all fields can be removed and let the factory method just return an initialized serializer.